### PR TITLE
[JBPM-9229] hasRole implementation shouldn't return always false

### PR DESCRIPTION
**Pull request by fjtirado**
A proper implementation for hasRole has been implemented

### CODE REVIEW

**Comment by fjtirado** [comment]
jenkins retest this

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/security/JACCIdentityProvider.java [hunk 1]
```diff
@@ -91,6 +92,31 @@ public String getName() {
 
     @Override
     public boolean hasRole(String s) {
+        Subject subject = getSubjectFromContainer();
+        if (subject != null) {
+        Set<Principal> principals = subject.getPrincipals();
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/security/JACCIdentityProvider.java:97]: Missing indentation here

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/security/JACCIdentityProvider.java [hunk 2]
```diff
@@ -91,6 +92,31 @@ public String getName() {
...
+            if (principals != null) {
+                for (Principal principal : principals) {
+                    if (principal instanceof Group) {
+                        Enumeration<? extends Principal> groups = ((Group) principal).members();
+                        while (groups.hasMoreElements()) {
+                            Principal groupPrincipal = (Principal) groups.nextElement();
+                            if (s.equals(groupPrincipal.getName())) {
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/security/JACCIdentityProvider.java:104]: Avoid possible NullPointerException in case "s" is null:
```suggestion
                            if (groupPrincipal.getName().equals(s)) {
```

File: kie-server-parent/kie-server-services/kie-server-services-common/src/test/java/org/kie/server/services/impl/security/JACCIdentityProviderWildFlyTomcatTest.java [hunk 3]
```diff
@@ -101,6 +102,9 @@ public void testGetRoles() throws Exception {
         assertEquals(2, jaccIdentityProvider.getRoles().size());
         assertTrue(jaccIdentityProvider.getRoles().contains(GROUP_ONE_NAME));
         assertTrue(jaccIdentityProvider.getRoles().contains(GROUP_TWO_NAME));
+        assertTrue(jaccIdentityProvider.hasRole(GROUP_ONE_NAME));
+        assertTrue(jaccIdentityProvider.hasRole(GROUP_TWO_NAME));
+        assertFalse(jaccIdentityProvider.hasRole("non_existing_role"));
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-services/kie-server-services-common/src/test/java/org/kie/server/services/impl/security/JACCIdentityProviderWildFlyTomcatTest.java:107]: You can add a test for null role:
`assertFalse(jaccIdentityProvider.hasRole(null));`

**Comment by fjtirado** [comment]
jenkins retest this

**Comment by fjtirado** [comment]
jenkins retest this

**Comment by fjtirado** [comment]
jenkins retest this

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/security/JACCIdentityProvider.java [hunk 4]
```diff
@@ -91,6 +92,31 @@ public String getName() {
...
     public boolean hasRole(String s) {
+        Subject subject = getSubjectFromContainer();
+        if (subject != null) {
+            Set<Principal> principals = subject.getPrincipals();
+            if (principals != null) {
+                for (Principal principal : principals) {
+                    if (principal instanceof Group) {
+                        Enumeration<? extends Principal> groups = ((Group) principal).members();
+                        while (groups.hasMoreElements()) {
+                            Principal groupPrincipal = (Principal) groups.nextElement();
```
Comment (afalhambra) [inline kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/security/JACCIdentityProvider.java:103]: Cast to `Principal` is redundant
```suggestion
                            Principal groupPrincipal = groups.nextElement();
```

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/security/JACCIdentityProvider.java [hunk 5]
```diff
@@ -91,6 +92,31 @@ public String getName() {
...
+                            if (groupPrincipal.getName().equals(s)) {
```
Comment (afalhambra) [inline kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/security/JACCIdentityProvider.java:104]: Not really sure here, should we ignore case sensitive? What do you think?

Comment (fjtirado) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/security/JACCIdentityProvider.java:104]: hmmm, good point, since there are not requirement about that, lets keep it this way, which is the default.

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/security/JACCIdentityProvider.java (hunk 4 above)
Comment (fjtirado) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/security/JACCIdentityProvider.java]: good catch, removing this casting from here and former code

**Comment by afalhambra** [comment]
Jenkins retest this please

**Comment by fjtirado** [comment]
jenkins retest this

**Comment by fjtirado** [comment]
jenkins retest this

**Comment by fjtirado** [comment]
@kiegroup/gatekeepers please merge
